contracts/JBV3Token.sol

```diff
@@ -0,0 +1,355 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol';
+import '@jbx-protocol-v2/contracts/interfaces/IJBToken.sol';
+import '@jbx-protocol-v2/contracts/interfaces/IJBController.sol';
+import '@jbx-protocol-v2/contracts/interfaces/IJBDirectory.sol';
+import '@jbx-protocol-v2/contracts/interfaces/IJBTokenStore.sol';
+import '@jbx-protocol-v1/contracts/interfaces/ITicketBooth.sol';
+
+/** 
+  @notice
+  An ERC-20 token that can be used by a project in the `JBTokenStore`.
+
+  @dev
+  Adheres to -
+  IJBToken: Allows this contract to be used by projects in the JBTokenStore.
+
+  @dev
+  Inherits from -
+  ERC20Votes: General token standard for fungible membership with snapshot capabilities sufficient to interact with standard governance contracts. 
+  Ownable: Includes convenience functionality for checking a message sender's permissions before executing certain transactions.
+*/
+contract JBV3Token is ERC20Votes, Ownable, IJBToken {
+  //*********************************************************************//
+  // --------------------------- custom errors ------------------------- //
+  //*********************************************************************//
+  error BAD_PROJECT();
+  error INSUFFICIENT_FUNDS();
+
+  //*********************************************************************//
+  // --------------------- public stored properties -------------------- //
+  //*********************************************************************//
+
+  /** 
+    @notice
+    The ID of the project that this token should be exclusively used for. Send 0 to support any project. 
+  */
+  uint256 public immutable projectId;
+
+  /** 
+    @notice
+    The V1 Token Booth Instance. 
+  */
+  ITicketBooth public immutable v1TicketBooth;
+
+  /** 
+    @notice
+    The V2 Token Store Instance. 
+  */
+  IJBTokenStore public immutable v2TokenStore;
+
+
+  /** 
+    @notice
+    The V3 Directory Instance. 
+  */
+  IJBDirectory public immutable v3Directory;
+
+  //*********************************************************************//
+  // ------------------------- external views -------------------------- //
+  //*********************************************************************//
+
+  /** 
+    @notice
+    The total supply of this ERC20.
+
+    @param _projectId the ID of the project to which the token belongs. This is ignored.
+
+    @return The total supply of this ERC20, as a fixed point number.
+  */
+  function totalSupply(uint256 _projectId) external view override returns (uint256) {
```

Comment:
Should we override erc20 totalSupply() too?

Reply:
`erc20 votes` doesn't overrides it https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Votes.sol#L27 so should be good I think wdyt ?

Reply:
Yh, it's rather for ui/etherscan, as we woudl return 2 different totalSupply (with or without previous versions)